src/legacy/ui/apm/index.js

```diff
@@ -30,22 +30,21 @@ export function apmInit(config) {
   return apmEnabled ? `init(${config})` : '';
 }
 
-export function getApmConfig(appMetadata) {
+export function getApmConfig(app) {
   if (!apmEnabled) {
-    return {};
+    return null;
   }
   /**
    * we use the injected app metadata from the server to extract the
-   * app URL path to be used for page-load transaction
+   * app id to be used for page-load transaction
    */
-  const navLink = appMetadata.getNavLink();
-  const pageUrl = navLink ? navLink.toJSON().url : appMetadata._url;
+  const appId = app.getId();
```

Comment:
In the NP this will always be `core`, so I'm not sure that's very useful?

Reply:
Yes, I have left a comment in the description on why its like this for now. 

> Since we don't have a nice way to get the currently navigated app after the Kibana is bootstrapped, we will use the app.getId() which would return `core` for NP and consolidate all transactions under core. Issue to track this #53466

Reply:
Perhaps add a comment about this in the code as well.

Reply:
Sure, I will add it on my upcoming PR which I am experimenting with using `window.location.href` since we are using it only for dev environment and can use it till we get a solution in NP.